tiledb/sm/cpp_api/array.h

```diff
@@ -708,12 +710,16 @@ class Array {
    *
    * @param ctx The TileDB context.
    * @param uri The array URI.
+   * @param config The optional request for additional features.
    * @return The loaded ArraySchema object.
    */
-  static ArraySchema load_schema(const Context& ctx, const std::string& uri) {
+  static ArraySchema load_schema(
+      const Context& ctx,
+      const std::string& uri,
+      const Config& config = Config()) {
     tiledb_array_schema_t* schema;
-    ctx.handle_error(
-        tiledb_array_schema_load(ctx.ptr().get(), uri.c_str(), &schema));
+    ctx.handle_error(tiledb_array_schema_load_with_config(
```

Comment:
After I did the overload it did occur to be that something like a function pointer would probably break.  Breaking a Cython declaration (for example) was something I didn't think of.

> maybe we could get the config from the Context so that both APIs load enumerations on schema load

The config is deliberately cleared here
```
  // Use a default constructed config to load the schema with default options.
  *array_schema = tiledb_array_schema_t::make_handle(
      load_array_schema(ctx->context(), sm::URI(array_uri), sm::Config()));
```
in commit 9b4e5ea0c8

was there a specific configuration option that we wanted to avoid here?  Is it your recommendation to change this instead to `ctx->context()->config()`?

Reply:
Good point, I'm not sure how destructive that would be on CI now that you mention it.. It could be that using the context configuration in `tiledb_array_schema_load` I ran into a lot of failures so I went for a totally separate `_with_config` API instead. I was just pointing out the potential breakage for tiledb-py and trying to think of a way around that. Feel free to give it a shot if you'd like, but if using the context configurations causes too much CI troubles I'd say this is fine too if it works for you

```c++
  static ArraySchema load_schema(
      const Context& ctx,
      const std::string& uri) {
    tiledb_array_schema_t* schema;
    ctx.handle_error(tiledb_array_schema_load(
        ctx.ptr().get(), uri.c_str(), &schema));
    return ArraySchema(ctx, schema);
  }

  static ArraySchema load_schema_with_config(
      const Context& ctx,
      const std::string& uri,
      const Config& config) {
    tiledb_array_schema_t* schema;
    ctx.handle_error(tiledb_array_schema_load_with_config(
        ctx.ptr().get(), config.ptr().get(), uri.c_str(), &schema));
    return ArraySchema(ctx, schema);
  }
```

Reply:
Nah I'm taking the easy route here.  754f239e undoes the default argument and adds the separate named option as suggested.